### PR TITLE
[38690] Email design - Daily Reminders

### PR DESCRIPTION
**Pull request by HDinger**
### ToDo

- [x] The daily email should contain all unread notifications that happened since last time you received a daily reminder.
- [x] The header should include:
  - [x] Logo of Open project.
  - [x] Salutation with the name of the user.
  - [x] Information: When this email is generated (for search reasons), “X” unread notifications, in “X” Work Packages.
  - [x] A button access to the notification centre (main view with no work package selected).
- [x] The content should be divided in notification rows as in the Notification Centre (with some UI variations as card shadows).
  - [x] Aggregate all notifications for one Work Package 
  - [x] In this case all the lines of changes will be displayed (if there is 3 notifications in a single Work Package each of them will be displayed in a line).
  - [x] Show configured colours for type and status
  - [x] When the user clicks in one of the rows the Notification Centre opens with the specific activity for that work package open.
  - [x] Also update text only version to match the new styled email
- [x] A button with a link to the email settings is displayed at the end of the email.
- [x] A limit of 15 work packages will be applied to this mails. If the user has more than 15 work packages with notifications a "show all work packages" button will be displayed that will lead to the Notification Centre main view.
- [x] Check for small screens
- [x] Adapt tests

https://community.openproject.org/projects/openproject/work_packages/38690/activity

### CODE REVIEW

**Review by ulferts — CHANGES_REQUESTED** [review]
I checked the resulting mail with the following clients:
* Thunderbird on mac
* Mail on mac
* mailbox.org UI using Chrome on mac
* Mail on iPhone

When reviewing I did not have access to the figma files yet, so I am using the screenshots shown in https://community.openproject.org/wp/38690 as a reference. 

Some of the deviations I find acceptable personally and could understand that judging them by an effort vs. gain equation might result in them not being fixed.

# Thunderbird on mac

* The logo is not displayed at the top:
 
image

From what I see in the code, that icon would always be the OpenProject icon. This might not be desired by every customer if that customer has uploaded a custom icon. Admittedly, this has not been specified.

* There are no icons in the "notification center" button and in the "Change email settings" button. This seems to be accepted as Out of scope, though.
image
image


* The 'id - project - reasons' line is not printed in bold
image

* No drop shadow around each notification (after having received access to figma, this can probably be dropped since there are no shadows in figma)
image

* If the mail window's width is decreased, and longer lines are present in the notifications and the user then scrolls to the right, lines breaking out of the notification boxes become visible

image

* The notification settings will be split up into two pages (notification settings and email reminders). The link on the "Change email settings" button currently points to the notification settings which is ok since the other does not exist yet. But either the wording or the target should be changed in the long run.

* The logo is embedded at the bottom of the mail:
image
 
* The subject in the screenshot has a different format: OpenProject (or instance name?) - x unread notifications in y work packages, in z projects
 
image

* The margins between boxes and at various other places (e.g. between the horizontal row and the first notification box) seem to be less than in the screenshot.

# Mail on mac

* Because of the dark mode, parts of the mail are very hard to read
image
image
image



* The logo is embedded at the bottom (but differently) and missing at the top
image

# mailbox.org UI using Chrome on mac

* No rounded button for the "Notification center" link (the button to "Change email settings" is rounded) 
image

* Increased height for the status badge
* Id written vertically
* Type written vertically
image

# Mail on iPhone

* Inconsistent font size between notifications
* notification count pill not floating right
image

File: app/helpers/mail_notification_helper.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+module MailNotificationHelper
+  include ::ColorsHelper
+
+  def logo_tag(**options)
+    current_logo = CustomStyle.current.logo
```
Comment (ulferts) [inline app/helpers/mail_notification_helper.rb]: `CustomStyle.current` might be nil resulting in a an error when sending `logo` to it.

**Review by ulferts — CHANGES_REQUESTED** [review]
# Contents regardless of client

* The reasons for notification use different wording compared to the Figma file
e.g.
image

vs 

image
The Figma file seems to expect to differentiate between "Assignee" and "Accountable" as well which is out of scope for this PR

* The headings for the project are no longer specified to be bold. This seems to have been changed in the meantime.

* The subject line does not note down the number or projects: e.g. "23 unread notification(s) in 8 work package(s)" 

* The link "Change email settings" should point to the "reminder settings" (clarified only recently)

* The "status badges" in the Figma file have more padding to the left and right compared to the implementation.
Mail:
image
Figma:
image

# Thunderbird on mac 

* The details sometimes break out of the boxes when decreasing the width of the mail window

image



# Mail on mac 

Same problems as in Thunderbird

# Mailbox.org UI in Chrome on Mac

Same problems as in Thunderbird

# Mail on iOS

* Different font sizes between notification rows

image

This might be something iOS does in order to increase readability but that is just guessing.

* Personal preference: I don't like the way long headings caused by project names are displayed on mobile devices. To me, they should be displayed in their own row but that would require media queries and I don't assume them to be supported.

image

# GMX in Chrome on Mac

* rounded corners are not supported at all. I think the outcome is acceptable, though.

image

# Gmail in Chrome on Mac

All fine. They seem to have implemented a min width.

**Comment by HDinger** [comment]
I have addressed most of the issues. There are of course still some limitations we have to live with (e.g. rounded corners are not supported by all clients). The concrete text for the subject and the header part is currently still discussed so I'd leave it like this for now and change it later separately if needed.

**Review by ulferts — APPROVED** [review]
Looking good design-wise now, @HDinger 🥇 🎈 👍

**Comment by HDinger** [comment]
Thanks a lot for your ongoing reviews @ulferts ! 👏 I'll merge this then once I got the tests green.

**Pull request merged**